lib/dao.js

```diff
@@ -1789,11 +1827,24 @@ DataAccessObject.prototype.save = function (options, cb) {
             });
           }
 
-          if (connector.save.length === 4) {
-            connector.save(modelName, inst.constructor._forDB(data), options, saveCallback);
-          } else {
-            connector.save(modelName, inst.constructor._forDB(data), saveCallback);
-          }
+          context = {
+            Model: Model,
+            data: data,
+            currentInstance: inst,
```

Comment:
But for `save()`, the `context` object being passed to both `before save` and `after save` hooks did not have `where`. So, I had skipped adding it here.